obs_maven/core.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, config_path, repo):
         self.url = data.get("url", "https://download.opensuse.org/repositories")
         self.repo = repo
         repositories = data.get("repositories", {})
-        repos = {name: Repo(self.url, data["project"], data["repository"]) for name, data in repositories.items()}
+        repos = {name: Repo(self.url, data.get("project"), data.get("repository"), data.get("url")) for name, data in repositories.items()}
```

Comment:
That also doesn't look good to me: why not filter out bogus data in comprehensions? Also I would not shadow `name` here as you don't need it anyways:
```python
repos = {name: Repo(self.url, data.get("project"), 
                              data.get("repository"),
                              data.get("url")) for data in repositories.values() if data.get("url")}
```

Downside: you are accessing `data` dict twice, but dicts are super-cheap in Python compiler by index, so it is quite OK here. Alternatively, you could use function `filter()` to prevent twice access. Filter is generally faster as it returns iterator (and so does `values()`), e.g.:

```python
repos = {name: Repo(......., filter(None, data.get("url") for data in repositories.values()))
```

It is very easy to achieve bogus data:

```yaml
artifacts:
   project:
```